src/cryptography/hazmat/bindings/openssl/binding.py

```diff
@@ -4,11 +4,25 @@
 
 from __future__ import absolute_import, division, print_function
 
+import os
 import threading
 
 from cryptography.hazmat.bindings._openssl import ffi, lib
 
 
+@ffi.callback("int (*)(unsigned char *, int)", error=-1)
```

Comment:
https://www.openssl.org/docs/crypto/RAND_bytes.html -1 seems more correct
than 0 (particularly for psueobytes)

On Sun, Jun 28, 2015 at 3:49 PM, Paul Kehrer <notifications@github.com>
wrote:

> In src/cryptography/hazmat/bindings/openssl/binding.py
> <https://github.com/pyca/cryptography/pull/2073#discussion_r33426214>:
>
> >  import threading
> >
> >  from cryptography.hazmat.bindings._openssl import ffi, lib
> >
> >
> > +@ffi.callback("int (*)(unsigned char *, int)", error=-1)
>
> Things will work fine without it, but I'm curious why we're returning -1
> here in case of error? 0 is what the old code returned.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/2073/files#r33426214>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084